src/modules/cmdpal/Microsoft.CmdPal.UI/Controls/SearchBar.xaml.cs

```diff
@@ -108,10 +111,20 @@ private void FilterBox_KeyDown(object sender, KeyRoutedEventArgs e)
 
             e.Handled = true;
         }
+        else if (e.Key == VirtualKey.Back)
+        {
+            // hack
+            if (CurrentPageViewModel != null)
+            {
+                CurrentPageViewModel.Filter = FilterBox.Text;
+            }
+        }
     }
 
     private void FilterBox_TextChanged(object sender, TextChangedEventArgs e)
```

Comment:
I thought the PR comment mentioned shortcutting the debouncer somewhere, where am I missing seeing that in the changes?

The debouncer issue you saw is mostly likely the bug that I detected and fixed as part of the improvements I made (including tests) there: https://github.com/CommunityToolkit/Windows/pull/569 - we've just been waiting on some other stuff to push a new update to NuGet out (hopefully in a day or two). Otherwise, you could pull from the `MainLatest` feed like we're doing for Labs.

Reply:
I just saw the comment in the PR and see that I missed it literally right above here in the `Back`space key handle. Yeah, I'm 99% sure this is the issue with the debouncer I fixed. If you hit Escape vs. backspace it should be fine as it's a different codepath.